examples/shell/nxp/k32w/k32w0/include/AppEvent.h

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ *    Copyright (c) 2021 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+struct AppEvent;
+typedef void (*EventHandler)(AppEvent *);
+
+struct AppEvent
+{
+    enum AppEventTypes
+    {
+        kEventType_Button = 0,
+        kEventType_Timer,
+        kEventType_TurnOn,
+        kEventType_Install,
+    };
+
+    uint16_t Type;
```

Comment:
should this be AppEventTypes rather than uint16_t?

Should the AppEventTypes be `enum class` or derive from uint16_t?

Reply:
Right, replaced uint16_t with AppEventTypes. Kept simple "enum" for AppEventTypes.